src/software/backend/simulation/simulator_robot.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <cinttypes>
+#include <optional>
+
+#include "software/backend/simulation/physics/physics_robot.h"
+extern "C"
+{
+#include "app/world/firmware_robot.h"
+}
+
+/**
+ * This class acts as a wrapper around a PhysicsRobot so that the PhysicsRobot
+ * can provide the interface of a FirmwareRobot.
+ *
+ * Because our firmware structs rely on C-style function pointers, we
+ * cannot use C++ constructs like std::function to provide to the structs.
+ * Therefore we use this class's static functions to provide the function
+ * pointers, since static functions work as C-style function pointers
+ * (because there is no instance associated with them).
+ *
+ * Whenever a caller needs to get a firmware struct or perform an operation, they
+ * need to set which robot they want to control so that all the static functions
+ * that have been provided to the firmware struct operate on the correct
+ * instantiated object. This is our workaround to maintain and simulate multiple
+ * "instances" of robot firmware at once.
+ */
+class SimulatorRobot
```

Comment:
We could call this `SimulatorRobotSingleton` to indicate that there is (effectively) only ever one instance?

Reply:
changed for robot and ball